### ISSUE TITLE
Windows Terminal Canary 1.23.3181.0 does not update

### ISSUE THREAD

**Issue by vbrozik** [microsoft/terminal#18204]
### Windows Terminal version

1.23.3181.0

### Windows build number

10.0.22631.4317

### Other Software

N/A

### Steps to reproduce

In Windows Terminal Canary 1.23.3181.0 open the About pop-up. The pop-up will show that an update is available:
![Image](https://github.com/user-attachments/assets/47fe75a9-291d-476c-8e27-3535a56828ca)

Close the terminal and restart it.

### Expected Behavior

The new running instance should be updated.

Note that it is user-unfriendly that the Terminal does not show any information about when the update will be installed and when it is ongoing.

### Actual Behavior

There is still the same version, again showing "An update is available."

I tried waiting between the individual steps (to allow the terminal to update) but the result is always the same.

The older Canary versions updated fine. I think that in some cases I had to wait between the steps but because of the lacking UI feedback I have no idea when should I wait and for how long.

**Comment by vbrozik** [comment]
Later I noticed that I accidentally opened the "Quake" terminal which was in the background all the time. So after I closed all Windows Terminal windows there still was an instance running. After exiting the Quake terminal the update finished successfully.

IMHO this illustrates the weaknesses of the current UI for updates. For the beginning it would be helpful to add a button with meaning: "close all instances allowing update to start".
